pysb/simulator/cupsoda.py

```diff
@@ -474,12 +474,13 @@ def _get_cmatrix(self):
             for j in range(self._len_rxns):
                 rate = 1.0
                 for r in rate_args[j]:
-                    x = str(r)
-                    if x in par_names:
-                        rate *= par_vals[i][par_dict[x]]
+                    if isinstance(r, pysb.Parameter):
+                        rate *= par_vals[i][par_names.index(r.name)]
```

Comment:
Was going through `par_dict` causing a problem, or did you decide it was just an unnecessary optimization?